pkg/metrics/metrics.jsx

```diff
@@ -1055,22 +1055,29 @@ class MetricsHour extends React.Component {
     }
 
     componentDidMount() {
-        this.updateGraphs(this.props.data, this.props.startTime);
+        this.updateGraphs(this.props.data, this.props.startTime, this.props.selectedVisibility);
     }
 
     shouldComponentUpdate(nextProps, nextState) {
-        if (this.state.dataItems !== nextProps.data.length || this.props.startTime !== nextProps.startTime) {
-            this.updateGraphs(nextProps.data, nextProps.startTime);
+        if (this.state.dataItems !== nextProps.data.length ||
+            this.props.startTime !== nextProps.startTime ||
+            Object.keys(this.props.selectedVisibility).some(itm => this.props.selectedVisibility[itm] != nextProps.selectedVisibility[itm])) {
+            this.updateGraphs(nextProps.data, nextProps.startTime, nextProps.selectedVisibility);
             return false;
         }
 
         return true;
     }
 
     // data: type → SAMPLES_PER_H objects from startTime
-    updateGraphs(data, startTime) {
+    updateGraphs(data, startTime, selectedVisibility) {
+        const filteredData = data.map(sample => Object.keys(sample)
+                .filter(key => {
+                    return selectedVisibility[key.split("_")[1]];
+                })
```

Comment:
This is better written as a pure arrow, i.e. `key => selectedVisibility[key.split("_")[1]]`